src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py

```diff
@@ -69,23 +88,22 @@ def firewall_rule_delete_func(client, resource_group_name, server_name, firewall
                 firewall_rule_name, server_name, resource_group_name))
     if confirm:
         try:
-            result = client.delete(resource_group_name, server_name, firewall_rule_name)
+            result = client.begin_delete(resource_group_name, server_name, firewall_rule_name)
         except Exception as ex:  # pylint: disable=broad-except
             logger.error(ex)
-    return result
+    return result.result()
```

Comment:
When `begin_delete` throw exception, is `result.result()` still needed? 

Is it better to have `result = client.begin_delete(...).result()` inside `try...except`?

Reply:
Exactly. When `LROPoller` is returned to Azure CLI core, there is no need to call `.result()` as the core will handle it for you.

Reply:
Made the change. Thanks a lot!!